File: backend/internal/dto/blood_request_request.go
package dto

type CreateBloodRequestRequest struct {
	PatientName     string   `json:"patient_name" binding:"required,max=100"`
	Relationship    string   `json:"relationship" binding:"required,max=50"`
	Location        string   `json:"location" binding:"required,max=150"`
	BloodType       string   `json:"blood_type" binding:"required,oneof=A B AB O"`
	Rhesus          string   `json:"rhesus" binding:"required,oneof=+ -"`
	BagsNeeded      int      `json:"bags_needed" binding:"required,min=1"`
	Urgency         string   `json:"urgency" binding:"required,oneof=Low Medium High Critical"`
	ContactPhone    string   `json:"contact_phone" binding:"required,max=20"`
	Notes           *string  `json:"notes" binding:"omitempty"`
}

type UpdateBloodRequestRequest struct {
	PatientName     string   `json:"patient_name" binding:"omitempty,max=100"`
	Relationship    string   `json:"relationship" binding:"omitempty,max=50"`
	Location        string   `json:"location" binding:"omitempty,max=150"`
	BloodType       string   `json:"blood_type" binding:"omitempty,oneof=A B AB O"`
	Rhesus          string   `json:"rhesus" binding:"omitempty,oneof=+ -"`
	BagsNeeded      *int     `json:"bags_needed" binding:"omitempty,min=1"`
	Urgency         string   `json:"urgency" binding:"omitempty,oneof=Low Medium High Critical"`
	ContactPhone    string   `json:"contact_phone" binding:"omitempty,max=20"`
	Notes           *string  `json:"notes" binding:"omitempty"`
}

File: mobile/lib/features/blood_request/data/models/blood_request_request.dart
import 'package:freezed_annotation/freezed_annotation.dart';

part 'blood_request_request.freezed.dart';
part 'blood_request_request.g.dart';

/// DTO untuk POST /api/v1/blood-requests
/// Digunakan untuk mengirim data ketika membuat blood request baru
@freezed
class BloodRequestRequest with _$BloodRequestRequest {
  const factory BloodRequestRequest({
    /// Nama pasien yang membutuhkan darah
    @JsonKey(name: 'patient_name')
    required String patientName,

    /// Hubungan dengan pasien (ayah, ibu, saudara, dll)
    required String relationship,

    /// Nama rumah sakit tempat pasien ditangani
    @JsonKey(name: 'hospital_name')
    required String hospitalName,

    /// Alamat rumah sakit
    @JsonKey(name: 'hospital_address')
    required String hospitalAddress,

    /// Latitude lokasi rumah sakit (-90 sampai 90)
    required double latitude,

    /// Longitude lokasi rumah sakit (-180 sampai 180)
    required double longitude,

    /// Jenis darah: A, B, AB, O
    @JsonKey(name: 'blood_type')
    required String bloodType,

    /// Rhesus darah: +, -
    required String rhesus,

    /// Jumlah kantong darah yang dibutuhkan (minimal 1)
    @JsonKey(name: 'bags_needed')
    required int bagsNeeded,

    /// Tingkat urgency: Low, Medium, High, Critical
    required String urgency,

    /// Catatan tambahan (opsional)
    String? notes,
  }) = _BloodRequestRequest;

  factory BloodRequestRequest.fromJson(Map<String, dynamic> json) =>
      _$BloodRequestRequestFromJson(json);
}
